src/v/cluster/rm_stm.cc

```diff
@@ -1884,8 +1886,10 @@ void rm_stm::apply_fence(model::record_batch&& b) {
     if (fence_it->second <= bid.pid.get_epoch()) {
         fence_it->second = bid.pid.get_epoch();
         _log_state.inflight[bid.pid] = 0;
+        _log_state.expiration.erase(bid.pid);
         if (version == rm_stm::fence_control_record_version) {
             _log_state.tx_seqs[bid.pid] = tx_seq.value();
+            track_tx(bid.pid, transaction_timeout_ms.value());
```

Comment:
this change looks unrelated to what i'd expect having read the commit message:

> cluster: Use log_state expiration instead of mem_state

Reply:
added a comment